app.js

```diff
@@ -0,0 +1,34 @@
+// Import dependencies
+const express = require('express');
+const morgan = require('morgan');
+
+const { db, MONGODB_URI } = require('./server/database');
+const routes = require('./server/routes');
+const app = express();
+const PORT = process.env.PORT || 3001;
+
+// Middleware
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+app.use(morgan('dev'));
+
+// REST API routes
+app.use(routes);
+
+// Error handler
+app.use(function (err, req, res, next) {
+  console.log('====== ERROR =======');
+  console.error(err.stack);
+  res.status(500);
+});
+
+db.connect(MONGODB_URI, { useNewUrlParser: true, useUnifiedTopology: true })
+  .then(resp => {
+    // Starting Server on successful databse connection
+    app.listen(PORT, () => {
+      console.log(`🌎  ==> REST API Server now listening on PORT ${PORT}!`);
+    });
+  })
+  .catch(err => {
+    console.log(`Error connecting to database: ${err}`);
+  });
```

Comment:
I made a card to pull the db out into its on module and then in here we can just do an async await db.start() or something like that - what do u think?